src/workspaces/list/WorkspaceStateCell.tsx

```diff
@@ -0,0 +1,152 @@
+import { Icon, IconId, Modal } from '@terra-ui-packages/components';
+import React, { ReactNode, useState } from 'react';
+import { Link } from 'src/components/common';
+import ErrorView from 'src/components/ErrorView';
+import { FirstParagraphMarkdownViewer } from 'src/components/markdown';
+import colors from 'src/libs/colors';
+import * as Style from 'src/libs/style';
+import { WorkspaceWrapper as Workspace } from 'src/workspaces/utils';
+
+const styles = {
+  tableCellContainer: {
+    height: '100%',
+    padding: '0.5rem 0',
+    paddingRight: '2rem',
+    borderTop: `1px solid ${colors.light()}`,
+  },
+  tableCellContent: {
+    height: '50%',
+    display: 'flex',
+    alignItems: 'center',
+  },
+};
+
+const WorkspaceDescriptionCell = (props: { description: unknown | undefined }) => (
+  <div style={styles.tableCellContent}>
+    {/* @ts-expect-error - FirstParagraphMarkdownViewer is not typed, so ts expects a value for renderers */}
+    <FirstParagraphMarkdownViewer
+      style={{
+        height: '1.5rem',
+        margin: 0,
+        ...Style.noWrapEllipsis,
+        color: props.description ? undefined : colors.dark(0.75),
+        fontSize: 14,
+      }}
+    >
+      {props.description?.toString() || 'No description added'}
+    </FirstParagraphMarkdownViewer>
+  </div>
+);
+
+interface WorkspaceStateCellProps {
+  workspace: Workspace;
+}
+
+export const WorkspaceStateCell = (props: WorkspaceStateCellProps): ReactNode => {
```

Comment:
Can you write unit tests for this code?

Reply:
They would just be a direct clone of the tests in `RenderedWorkspaces.test.ts`.

I could also move them out, and create the 'WorkspaceStateCell' components directly, but I'm not sure there's any value to that, since this component is only really intended to be used there.

SonarCloud shows a 90% coverage for this file.

The breakdown by-file, from SonarCloud:
* `RenderedWorkspaces` 62%: it's treating all of this file as new code. The gaps in coverage are not anything that was changed other than the tsx conversion, and 
* `useWorkspaceStatePolling` at 40%: a lot of this is because I translated over `useSingleWorkspaceDeletionPolling` to `useSingleWorkspaceStatePolling`, but I haven't switched the dashboard over yet, to use it.
* `WorkspacesListModals.ts` 0% - This shows total coverage at 0%, but new code is only 2 lines of that.
* `Environments.stories.tsx` 0% - Changes to a mock component...
* `ajax.js` 0% - this entire file is also at 0%, with only 2 lines of new code

So It looks like most of the reason for the coverage it's giving is that it's treating the ts -> tsx conversion as all new code, and the new method in `useWorkspaceStatePolling` that hasn't been wired in to the dashboard yet (which IIRC has tests for the correct polling behavior there).

Reply:
OK, thank you for taking a look!